library/modules/Gui.cpp

```diff
@@ -484,6 +484,12 @@ bool Gui::matchFocusString(std::string focusString, bool prefixMatch) {
     }) != currentFocusStrings.end();
 }
 
+static void push_dfhack_focus_string(dfhack_viewscreen *vs, std::vector<std::string> &focusStrings)
+{
+    auto name = vs->getFocusString();
+    focusStrings.push_back(name.empty() ? "dfhack" : "dfhack/" + name);
+}
+
 std::vector<std::string> Gui::getFocusStrings(df::viewscreen* top)
```

Comment:
note I reworked the logic in this function to allow focus strings to work at all for viewscreens without a specific handler (they were entering the second case and getting lost)